docs: updating README to the new flame bloc version (#1737)
flame-engine/flame: packages/flame_bloc/README.md

# Flame Bloc 🔥🧱

`flame_bloc` offers a simple and natural (as in similar to `flutter_bloc`) way to use blocs and
cubits inside a `FlameGame`.

## ⚠️  IMPORTANT ⚠️  ##
This README only covers the new (>= 1.4.0) API of flame bloc, for the previous deprecated API check

> [this](./DEPRECATED_README.md).

Reviewer: I don't think we need this, if they want to read an old version of the readme they can look at the tag for an old version.

For a migration guide from the previous API to the current one,

> [check this article](https://verygood.ventures/blog/flame-bloc-new-api).

Reviewer 2: 💯

## How to use

Lets assume we have a bloc that handles player inventory, first we need to make it available to our
components.

We can do that by using `FlameBlocProvider` component:

```dart
class MyGame extends FlameGame {
  @override
  Future<void> onLoad() async {
    await add(
      FlameBlocProvider<PlayerInventoryBloc, PlayerInventoryState>(
        create: () => PlayerInventoryBloc(),
        children: [
          Player(),
          // ...
        ],
      ),
    );
  }
}
```

> With that, the `Player` component will now have access to our bloc.

Reviewer 2: ```suggestion
With the above changes, the `Player` component will now have access to our bloc.
```

If more than one bloc needs to be provided, `FlameMultiBlocProvider` can be used in a similar fashion:

```dart
class MyGame extends FlameGame {
  @override
  Future<void> onLoad() async {
    await add(
      FlameMultiBlocProvider(
        providers: [
          FlameBlocProvider<PlayerInventoryBloc, PlayerInventoryState>(
            create: () => PlayerInventoryBloc(),
          ),
          FlameBlocProvider<PlayerStatsBloc, PlayerStatsState>(
            create: () => PlayerStatsBloc(),
          ),
        ],
        children: [
          Player(),
          // ...
        ],
      ),
    );
  }
}
```

> Listening to states changes on a component level can be done with two approaches:

Reviewer 2: ```suggestion
Listening to states changes at the component level can be done with two approaches:
```

By using `FlameBlocListener` component:

```dart
class Player extends PositionComponent {
  @override
  Future<void> onLoad() async {
    await add(
      FlameBlocListener<PlayerInventoryBloc, PlayerInventoryState>(
        listener: (state) {
          updateGear(state);
        },
      ),
    );
  }
}
```

Or by using `FlameBlocListenable` mixin:

```dart
class Player extends PositionComponent
  with FlameBlocListenable<PlayerInventoryBloc, PlayerInventoryState> {

  @override
  void onNewState(state) {
    updateGear(state);
  }
}

> ```

Reviewer 2: Might want to call out one limitation of the mixin is that a component cannot listen to multiple blocs
